xbmc/FileItem.cpp

```diff
@@ -2870,289 +2873,225 @@ void CFileItemList::Stack(bool stackFiles /* = true */)
-            // and recording last file item for later use
-            int nFiles = 0;
-            int index = -1;
-            for (int j = 0; j < items.Size(); j++)
+            CDirectory::GetDirectory(
+                item->GetPath(), items,
+                CServiceBroker::GetFileExtensionProvider().GetVideoExtensions(), DIR_FLAG_DEFAULTS);
+
+            // Find playable items
+            std::string playPath{};
+            std::string savePath{};
+            int playable{0};
+            for (const auto& dirItem : items)
             {
-              if (!items[j]->m_bIsFolder)
+              if (dirItem->m_bIsFolder)
+                // If subfolders then see if DVD/Bluray and find relevant VIDEO.TS/INDEX.BDMV
+                playPath = dirItem->GetOpticalMediaPath();
+              else
+                playPath = dirItem->GetPath();
+              if (!playPath.empty() && isVideoExtension.find(StringUtils::ToLower(
+                                           URIUtils::GetExtension(playPath))) != std::string::npos)
               {
-                nFiles++;
-                index = j;
+                savePath = playPath;
+                ++playable;
               }
-
-              if (nFiles > 1)
-                break;
             }
-
-            if (nFiles == 1)
-              *item = *items[index];
+            if (playable == 1)
+            {
+              // Only expect one playable item per folder
+              // Add to stack vector
+              stackCandidates.emplace_back(FOLDER_CANDIDATE, dirTitle, dirVolume,
+                                           &item - &m_items[0], savePath, item->m_dwSize);
+              break;
+            }
           }
-          ++expr;
         }
-
-        // check for dvd folders
-        if (!bMatch)
+      }
+      else if (!(item->m_bIsFolder || item->IsParentFolder() || item->IsNFO() ||
+                 item->IsPlayList()))
+      {
+        // File stacking
+        std::string file{};
+        std::string filePath{};
+        URIUtils::Split(StringUtils::ToLower(item->GetPath()), filePath, file);
+        if (URIUtils::HasEncodedFilename(CURL(filePath)))
+          file = CURL::Decode(file);
+
+        // Test each item against each RegExp
+        for (auto& regExp : fileRegExps)
         {
-          std::string dvdPath = item->GetOpticalMediaPath();
-
-          if (!dvdPath.empty())
+          if (regExp.RegFind(file) != -1)
           {
-            // NOTE: should this be done for the CD# folders too?
-            item->m_bIsFolder = false;
-            item->SetPath(dvdPath);
-            item->SetLabel2("");
-            item->SetLabelPreformatted(true);
-            m_sortDescription.sortBy = SortByNone; /* sorting is now broken */
+            // Get components of file name
+            const std::string fileTitle = regExp.GetMatch(1);
+            const std::string fileVolume = regExp.GetMatch(2);
```

Comment:
You don't really need these values as variables, as you need the data only once.

Reply:
It was just to make it more readable. I'll put a comment in instead.